credentialz/credentialz.proto

```diff
@@ -0,0 +1,605 @@
+  // reboots.
+  string version = 3;
+  // `created_on` contains information when the credentials were created.
+  // This information is controlled by the credential manager and reported as-is
+  // by the telemetry reporting system (ie, transparent to the device credential
+  // service).  Credential managers should use the timestamp of the moment when
+  // credential was created, not the time when the credential is pushed to
+  // a particular switch).
+  // Also, this timestamp should be persisted by the devices onto non-volatile
+  // memory for preservation across system reboots.
+  // `created_on` is a timestamp: the number of seconds since
+  // January 1st, 1970 00:00:00 GMT.
+  uint64 created_on = 4;
+}
+
+message RotateAuthorizedKeysResponse {
+}
+
+message RotateAuthorizedUsersRequest {
+  // A list of policies that are to be applied on the SSH service on
+  // the network element.
+  // Multiple policies MAY be specified to allow for multiple
+  // system roles, or specific user restrictions to be described.
+  repeated UserPolicy policies = 1;
+}
+
+message UserPolicy {
+  message SshAuthorizedUser {
+    string authorized_user = 1;
+    // Options specified for this authorized user.
+    repeated Option options = 2;
+  }
+
+  // The system role account name (e.g. root). This account must exist.
+  string account = 1;
+
+  // List of authorized usernames for this system role as found in the SSH
+  // certificate.
+  repeated SshAuthorizedUser authorized_users = 2;
+  // `version` contains versioning information that is controlled by
+  // the credential manager and reported as-is by the telemetry reporting system
+  // (ie, transparent to the device credential service).  Credential managers
+  // should choose version strings as discrete as possible to ease alert
+  // generation (eg, for credentials sourced from a bundle, the timestamp of
+  // the bundle should be used but not the time when the credential is pushed to
+  // a particular switch).  Also, such version strings should be persisted by
+  // the devices onto non-volatile memory for preservation across system
+  // reboots.
+  string version = 3;
+  // `created_on` contains information when the credentials were created.
+  // This information is controlled by the credential manager and reported as-is
+  // by the telemetry reporting system (ie, transparent to the device credential
+  // service).  Credential managers should use the timestamp of the moment when
+  // credential was created, not the time when the credential is pushed to
+  // a particular switch).
+  // Also, this timestamp should be persisted by the devices onto non-volatile
+  // memory for preservation across system reboots.
+  // `created_on` is a timestamp: the number of seconds since
+  // January 1st, 1970 00:00:00 GMT.
+  uint64 created_on = 4;
+}
+
+message RotateAuthorizedUsersResponse {
+}
+
+message RotatePasswordRequest {
+  message Account {
+    message Password {
```

Comment:
i don't like embeding a embedded message can we move that up?

Reply:
Done.